packages/fx-core/src/component/generator/officeXMLAddin/generator.ts

```diff
@@ -154,3 +156,86 @@ export class OfficeXMLAddinGenerator {
     return promisify(childProcess.exec)(cmdLine);
   }
 }
+
+export class OfficeXmlAddinGeneratorNew extends DefaultTemplateGenerator {
+  componentName = "office-xml-addin-generator";
+
+  public activate(context: Context, inputs: Inputs): boolean {
+    const projectType = inputs[QuestionNames.ProjectType];
+    const addinHost = inputs[QuestionNames.OfficeAddinHost];
+    return (
+      projectType === ProjectTypeOptions.officeXMLAddin().id &&
+      addinHost &&
+      addinHost !== OfficeAddinHostOptions.outlook().id
+    );
+  }
+
+  public async getTemplateInfos(
+    context: Context,
+    inputs: Inputs,
+    destinationPath: string,
+    actionContext?: ActionContext
+  ): Promise<Result<TemplateInfo[], FxError>> {
+    const host = inputs[QuestionNames.OfficeAddinHost] as string;
+    const capability = inputs[QuestionNames.Capabilities];
+    const lang = _.toLower(inputs[QuestionNames.ProgrammingLanguage]) as
```

Comment:
Is it possible that language is not TS/JS?

Reply:
Impossible for XML Add-in scenario.